tests/uk/co/ben_gibson/repositorymapper/RepositoryProvider/Host/Url/Factory/GitBlitTest.java

```diff
@@ -0,0 +1,162 @@
+    public void testGetUrlFromContext(Context context, String expectedUrl) throws MalformedURLException, UnsupportedEncodingException, RemoteNotFoundException, URISyntaxException, ProjectNotFoundException {
+        assertEquals(expectedUrl, this.getGitBlitUrlFactory().createUrl(context, false).toString());
+    }
+
+    /**
+     * Tests the url factory creates the correct url for a context.
+     */
+    @Test
+    @UseDataProvider("getContextsForSSL")
+    public void testGetUrlFromContextForcedSSL(Context context, String expectedUrl) throws MalformedURLException, UnsupportedEncodingException, RemoteNotFoundException, URISyntaxException, ProjectNotFoundException {
+        assertEquals(expectedUrl, this.getGitBlitUrlFactory().createUrl(context, true).toString());
+    }
+
+    /**
+     * Acts as a data provider for contexts.
+     *
+     * @return Object[][]
+     */
+    @DataProvider
+    public static Object[][] getContexts() throws MalformedURLException, RemoteNotFoundException {
+
+
+        return new Object[][]{
+                {
+                        ContextTestUtil.getMockedContext(
+                                "http://my.company.com/gitblit/git/category/my.complex.repo.git",
+                                "",
+                                "",
+                                "",
+                                "1234",
+                                32),
+                        "http://my.company.com/gitblit/commitdiff/category!my.complex.repo.git/1234"
+                },
+
+                {
+                        ContextTestUtil.getMockedContext(
+                                "http://git.delwink.com/git/r/patts-qt.git",
+                                "",
+                                "",
+                                "",
+                                "607a417df243451330d91b48515b4df9eb106580",
+                                32),
+                        "http://git.delwink.com/git/commitdiff/patts-qt.git/607a417df243451330d91b48515b4df9eb106580"
+                },
+                {
+                        ContextTestUtil.getMockedContext(
+                                "http://my.company.com/gitblit/git/category/my.complex.repo.git",
+                                "feature/feature42D",
+                                "src/com/foo/FooTest.java",
+                                "",
+                                "1234",
+                                32),
+                        "http://my.company.com/gitblit/blob/category!my.complex.repo.git/1234/src!com!foo!FooTest.java#L32"
+                },
+                {
+                        ContextTestUtil.getMockedContext(
+                                "http://git.delwink.com/git/r/ljpapi.git",
+                                "master",
+                                "src/com/delwink/ljp/Alignment.java",
+                                "",
+                                "607a417df243451330d91b48515b4df9eb106580",
```

Comment:
Are we missing a test for a context with no commit hash, which would produce a URL with the branch?